addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1357,81 +1357,111 @@ const ColorpickerUserValueWidget = SelectUserValueWidget.extend({
     },
 });
 
-const ImagepickerUserValueWidget = UserValueWidget.extend({
+const MediapickerUserValueWidget = UserValueWidget.extend({
     tagName: 'we-button',
     events: {
-        'click': '_onEditImage',
+        'click': '_onEditMedia',
     },
 
     /**
      * @override
      */
-    start: async function () {
+    async start() {
         await this._super(...arguments);
+        if (this.options.dataAttributes.buttonStyle) {
+            const iconEl = document.createElement('i');
+            iconEl.classList.add('fa', 'fa-fw', 'fa-camera');
+            this.containerEl.appendChild(iconEl);
+        } else {
+            this.el.classList.add('fa', 'fa-fw', 'fa-edit');
+        }
+    },
 
-        const allowedSelector = this.el.dataset.allowVideos;
-        this.allowVideos = allowedSelector ? this.$target.is(allowedSelector) : false;
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
 
-        const iconEl = document.createElement('i');
-        iconEl.classList.add('fa', 'fa-fw', 'fa-camera');
-        this.containerEl.appendChild(iconEl);
+    /**
+     * Creates and opens a media dialog to edit a given element's media.
+     *
+     * @private
+     * @param {HTMLElement} el the element whose media should be edited
+     * @param {boolean} [images] whether images should be available
+     *   default: false
+     * @param {boolean} [videos] whether videos should be available
+     *   default: false
+     */
+    _openDialog(el, {images = false, videos = false}) {
+        el.src = this._value;
+        const $editable = this.$target.closest('.o_editable');
+        const mediaDialog = new weWidgets.MediaDialog(this, {
+            noImages: !images,
+            noVideos: !videos,
+            noIcons: true,
+            noDocuments: true,
+            isForBgVideo: true,
+            'res_model': $editable.data('oe-model'),
+            'res_id': $editable.data('oe-id'),
+        }, el).open();
+        return mediaDialog;
     },
     /**
      * @override
      */
-    getMethodsParams: function (methodName) {
-        return _.extend({isVideo: this.isVideo}, this._super(...arguments));
+    async _updateUI() {
+        this._super(...arguments);
+        this.el.classList.toggle('active', this.isActive());
```

Comment:
@sdegueldre I push-forced the branch so don't touch it anymore please :) I reviewed the two first commits in details, I keep the big one for this week-end/monday.

- I added a commit of mine before your work: removing check icon for we-button active state
- I changed the design to not look like a big button, just the fa-edit icon
- I restored the blog option: the active state was lost because you remove the 'active' class addition here
- I changed the hierarchy to add spaces for multiple level hierarchy in options (we'll see with foldable sections later anyway but that way it is already more clear to me)